File: src/main/java/net/draconia/askaround/domain/AssetStatus.java
package net.draconia.askaround.domain;

import java.io.Serializable;

public class AssetStatus implements Serializable
{
	private static final long serialVersionUID = -5641828564465821589L;
	
	private Integer miId;
	private String msDescription, msStatus;
	
	public AssetStatus()
	{ }
	
	public AssetStatus(final String sStatus)
	{
		setStatus(sStatus);
	}
	
	public AssetStatus(final Integer iId, final String sStatus)
	{
		setId(iId);
		setStatus(sStatus);
	}
	
	public AssetStatus(final String sStatus, final String sDescription)
	{
		setStatus(sStatus);
		setDescription(sDescription);
	}
	
	public AssetStatus(final Integer iId, final String sStatus, final String sDescription)
	{
		setId(iId);
		setStatus(sStatus);
		setDescription(sDescription);
	}
	
	public String getDescription()
	{
		if(msDescription == null)
			msDescription = "";
		
		return(msDescription);
	}
	
	public int getId()
	{
		if(miId == null)
			miId = 0;
		
		return(miId);
	}
	
	public String getStatus()
	{
		if(msStatus == null)
			msStatus = "";
		
		return(msStatus);
	}
	
	public void setDescription(final String sDescription)
	{
		if(sDescription == null)
			msDescription = "";
		else
			msDescription = sDescription;
	}
	
	public void setId(final Integer iId)
	{
		if(iId == null)
			miId = 0;
		else
			miId = iId;
	}
	
	public void setStatus(final String sStatus)
	{
		if(sStatus == null)
			msStatus = "";
		else
			msStatus = sStatus;
	}
}

File: src/main/webapp/scripts/models/assetstatus.js
function AssetStatus()
{
	this.mbChanged						=	undefined;
	this.msDescription					=	undefined;
	this.miId							=	undefined;
	this.mArrObjObservers				=	[];
	this.msStatus						=	undefined;
};

AssetStatus.prototype.addObserver		=	function(objObserver)
{
	if(objObserver != undefined)
		{
		if(this.mArrObjObservers == undefined)
			this.mArrObjObservers 		=	[];
		
		this.mArrObjObservers.push(objObserver);
		}
	
	return(this);
};

AssetStatus.copyFromJSON				=	function(objJSON, objAssetStatus)
{
	if(objAssetStatus == undefined)
		objAssetStatus					=	new AssetStatus();
	
	for(prop in objJSON)
		objAssetStatus["set" + prop.substring(0, 1).toUpperCase() + prop.substring(1)].call(objAssetStatus, eval("objJSON." + prop));
	
	return(objAssetStatus);
};

AssetStatus.prototype.getDescription	=	function()
{
	if(this.msDescription == undefined)
		this.msDescription = "";
	
	return(this.msDescription);
};

AssetStatus.prototype.getId				=	function()
{
	if(this.miId == undefined)
		this.miId = 0;
	
	return(this.miId);
};

AssetStatus.prototype.getStatus			=	function()
{
	if(this.msStatus == undefined)
		this.msStatus = "";
	
	return(this.msStatus);
};

AssetStatus.prototype.hasChanged				=	function()
{
	if(this.mbChanged == undefined)
		this.mbChanged = false;
	
	return(this.mbChanged);
};

AssetStatus.prototype.notifyObservers			=	function(objArgument)
{
	this.mbChanged = true;
	
	for(var iLength = this.mArrObjObservers.length, iLoop = 0; iLoop < iLength; iLoop++)
		this.mArrObjObservers[iLoop].call(this, objArgument);
	
	this.mbChanged = false;
};

AssetStatus.prototype.packageToJSON		=	function()
{
	return(	{	id:				this.getId()
			,	status:			this.getStatus()
			,	description:	this.getDescription()
			});
};

AssetStatus.prototype.setDescription	=	function(sDescription)
{
	if(sDescription == undefined)
		this.msDescription = "";
	else
		this.msDescription = sDescription;
	
	this.notifyObservers();
	
	return(this);
};

AssetStatus.prototype.setId				=	function(iId)
{
	if(iId == undefined)
		this.miId = 0;
	else
		this.miId = iId;
	
	this.notifyObservers();
	
	return(this);
};

AssetStatus.prototype.setStatus			=	function(sStatus)
{
	if(sStatus == undefined)
		this.msStatus = "";
	else
		this.msStatus = sStatus;
	
	this.notifyObservers();
	
	return(this);
};